atlasdb-cassandra/src/main/java/com/palantir/atlasdb/keyvalue/cassandra/EntryPair.java

```diff
@@ -0,0 +1,94 @@
+/**
+ * Copyright 2017 Palantir Technologies
+ *
+ * Licensed under the BSD-3 License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://opensource.org/licenses/BSD-3-Clause
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.palantir.atlasdb.keyvalue.cassandra;
+
+import java.util.Arrays;
+import java.util.Optional;
+import java.util.UUID;
+
+import org.apache.cassandra.thrift.CASResult;
+import org.apache.cassandra.thrift.Column;
+import org.immutables.value.Value;
+
+import com.palantir.timestamp.MultipleRunningTimestampServiceError;
+
+@Value.Immutable
+public abstract class EntryPair {
+    abstract TimestampBoundStoreEntry noId();
+    abstract TimestampBoundStoreEntry withId();
+
+    public static EntryPair create(TimestampBoundStoreEntry withoutId, TimestampBoundStoreEntry withId) {
+        return ImmutableEntryPair.builder()
+                .noId(withoutId)
+                .withId(withId)
+                .build();
+    }
+
+    public static EntryPair createForTimestampAndId(long ts, UUID id) {
+        return create(TimestampBoundStoreEntry.create(ts, null), TimestampBoundStoreEntry.create(ts, id));
+    }
+
+    public static EntryPair createFromCasResult(CASResult result) {
+        if (result.getCurrent_values() == null || result.getCurrent_values().size() != 2) {
+            throw new MultipleRunningTimestampServiceError("Unexpected number of entries in the timestamp bound DB: "
+                    + result.getCurrent_values().size() + ". Expected number of entries is 2."
+                    + " This may indicate another timestamp service is running against this cassandra keyspace.");
+        }
+        Column firstColumn = result.getCurrent_values().get(0);
+        Column secondColumn = result.getCurrent_values().get(1);
+        EntryPair entries;
+        if (noIdColumn(firstColumn) && withIdColumn(secondColumn)) {
+            entries = createFromColumns(firstColumn, secondColumn);
+        } else {
+            entries = createFromColumns(secondColumn, firstColumn);
+        }
```

Comment:
Okay, I see that this is done because columns could be in arbitrary order. I think we should extract this to a separate method?